### PR TITLE
Parser improvements related to line break stripping

### PR DESCRIPTION
**Pull request by Aivean**
Changes
* strip line breaks at the beginning and the end of the text
* strip extra line breaks after the blockquote

Small demo:
orb-demo-bq

Testing:
* unit tests
* manual

### CODE REVIEW

**Comment by Aivean** [comment]
Fix #131

File: backend/src/parser/TheParser.ts [hunk 1]
```diff
@@ -54,14 +57,25 @@ export default class TheParser {
...
-        return doc.reduce((p, c) => {
-            const res = this.parseNode(c);
+        const p = {text: '', mentions: [], urls: [], images: []};
+        let prevBQ = false; // if previous node was blockquote
+        for (let node of doc) {
+            if (prevBQ && node.type === 'text') {
+                // remove a single newline after blockquote, allow only a single one if multiple were present
+                node = {
+                    ...node,
+                    data: node.data.replace(/^(\r?\n|[\r\n])(\r?\n|[\r\n])?(\r?\n|[\r\n])*/, '$2')
```
Comment (pazoozoo42) [inline backend/src/parser/TheParser.ts:67]: I think this *or* condition is redundant here: `|[\r\n]`, and if I change pattern to 
```js
data: node.data.replace(/^(\r?\n)(\r?\n)?(\r?\n)*/, '$2')
```
tests are passing too

**Review by pazoozoo42 — APPROVED** [review]
looks good, one minor suggestion to simplify the regex

File: backend/src/parser/TheParser.ts (hunk 1 above)
Comment (Aivean) [reply backend/src/parser/TheParser.ts:67]: Sure, but it doesn't account for `\r` line break type (not sure if we need to support it, but we do in other places).
https://en.wikipedia.org/wiki/Newline

Comment (Aivean) [reply backend/src/parser/TheParser.ts:67]: And now that I looked into it, my regex doesn't account for `\n\r` 🤷‍♂️

Comment (megamk) [reply backend/src/parser/TheParser.ts:67]: > Sure, but it doesn't account for \r line break type

maybe add tests with `\r` then

Comment (pazoozoo42) [reply backend/src/parser/TheParser.ts:67]: > Sure, but it doesn't account for \r line break type 

maybe add tests with `\r` then

**Pull request merged**